tests/bwc/test_upgrade.py

```diff
@@ -446,3 +456,118 @@ def start_cluster_and_alter_tables(self, version_def, nodes):
         self._process_on_stop()
+
+
+class SnapshotCompatibilityTest(NodeProvider, unittest.TestCase):
+
+    CREATE_REPOSITORY = '''
+CREATE REPOSITORY r1 TYPE S3
+WITH (access_key = 'minio',
+secret_key = 'miniostorage',
+bucket='backups',
+endpoint = '127.0.0.1:9000',
+protocol = 'http')
+'''
+
+    CREATE_SNAPSHOT_TPT = "CREATE SNAPSHOT r1.s{} ALL WITH (wait_for_completion = true)"
+
+    RESTORE_SNAPSHOT_TPT = "RESTORE SNAPSHOT r1.s{} ALL WITH (wait_for_completion = true)"
+
+    DROP_DOC_TABLE = 'DROP TABLE t1'
+
+    SUPPORTED_VERSION = ('3.3.x', '4.0')
+
+    def test_snapshot_compatibility(self):
+        """Test snapshot compatibility when upgrading 3.3.x -> 4.0.
+
+        Using Minio as a S3 repository, the first cluster that runs
+        creates the repo, a table and inserts/selects some data, which
+        then is snapshotted and deleted. The next cluster recovers the
+        data from the last snapshot, performs further inserts/selects,
+        to then snapshot the data and delete it.
+
+        We are interested in the transition 3.3.x -> 4.0
+        """
+        with MinioServer() as minio:
+            t = threading.Thread(target=minio.run)
+            t.daemon = True
+            t.start()
+            wait_until(lambda: _is_up('127.0.0.1', 9000))
+
+            num_nodes = 3
+            num_docs = 30
+            prev_version = None
+            num_snapshot = 1
+            path_data = 'data_test_snapshot_compatibility'
+            cluster_settings = {
+                'cluster.name': gen_id(),
+                'path.data': path_data
+            }
+            shutil.rmtree(path_data, ignore_errors=True)
+            for version in self.SUPPORTED_VERSION:
+                cluster = self._new_cluster(version, num_nodes, None, settings=cluster_settings)
+                cluster.start()
+                with connect(cluster.node().http_url, error_trace=True) as conn:
+                    c = conn.cursor()
+                    if not prev_version:
+                        c.execute(self.CREATE_REPOSITORY)
+                        c.execute(CREATE_ANALYZER)
+                        c.execute(CREATE_DOC_TABLE)
+                        insert_data(conn, 'doc', 't1', num_docs)
+                    else:
+                        c.execute(self.RESTORE_SNAPSHOT_TPT.format(num_snapshot - 1))
+                    c.execute('SELECT COUNT(*) FROM t1')
+                    rowcount = c.fetchone()[0]
+                    self.assertEqual(rowcount, num_docs)
+                    run_selects(c, version)
+                    c.execute(self.CREATE_SNAPSHOT_TPT.format(num_snapshot))
+                    c.execute(self.DROP_DOC_TABLE)
+                self._process_on_stop()
+                prev_version = version
+                num_snapshot += 1
+            shutil.rmtree(path_data, ignore_errors=True)
+
+
+class SnapshotHeterogeneousNodesCompatibilityTest(SnapshotCompatibilityTest):
+
+    SUPPORTED_VERSION = ('4.0', '4.0.8')
```

Comment:
```suggestion
    VERSIONS = ('4.0', '4.0.8')
```

Other versions would also be supported, we just don't test them.

Reply:
I think either the commit message or inline here as comment on the `VERSIONS` attribute we should clarify **why** we're testing these versions specificly.